apps/src/netsim/NetSimConnection.js

```diff
@@ -173,34 +176,24 @@ NetSimConnection.prototype.disconnectFromShard = function () {
  * @private
  */
 NetSimConnection.prototype.createMyClientNode_ = function (displayName) {
-  NetSimLocalClientNode.create(this.shard_, function (node) {
-    if (node) {
+  NetSimLocalClientNode.create(this.shard_, function (err, node) {
+    if (err === null) {
       this.myNode = node;
-      this.myNode.onChange.register(this.onMyNodeChange_.bind(this));
       this.myNode.setDisplayName(displayName);
+      this.myNode.setLostConnectionCallback(this.disconnectFromShard.bind(this));
       this.myNode.initializeSimulation(this.sentLog_, this.receivedLog_);
-      this.myNode.update(function () {
+      this.myNode.update(function (/*err, result*/) {
         this.shardChange.notifyObservers(this.shard_, this.myNode);
         this.statusChanges.notifyObservers();
       }.bind(this));
     } else {
+      logger.error(err.message);
       logger.error("Failed to create client node.");
```

Comment:
I would structure this:
```
if (err) { // or if (err !== null)
  logger.error()
  return;
}
// success code
```
Reduces nesting and means you can stop thinking about errors as you're past that first block.